internal/webhook/proxmoxmachine_webhook.go

```diff
@@ -53,6 +53,12 @@ func (p *ProxmoxMachine) ValidateCreate(_ context.Context, obj runtime.Object) (
 		return warnings, apierrors.NewBadRequest(fmt.Sprintf("expected a ProxmoxMachine but got %T", obj))
 	}
 
+	err = validateLocalStorage(machine)
+	if err != nil {
+		warnings = append(warnings, "localStorage is mutually exclusive with templateID/sourceNode")
+		return warnings, apierrors.NewBadRequest("localStorage is mutually exclusive with templateID/sourceNode")
```

Comment:
Since you can't see the sonarqube report (sorry about that): it's complaining about this string constant being repeated 5 times. Personally, I don't really see the point in setting warnings here at all only to return the same warning again as an error. This also applies to ValidateUpdate below. validateLocalStorage should return an error that can be returned by ValidateCreate and ValidateUpdate directly.